docs/index.rst

```diff
@@ -1406,6 +1444,67 @@ It is possible to use factories in
     For the same reason as above, it is not possible to use `Factories as Services`_ with required
     constructor arguments (the container is not yet available).
 
+.. tip::
+
+    ``ModelFactory::new()->many()`` and ``ModelFactory::new()->sequence()`` return a special object ``FactoryCollection``
+    which can be used to generate data providers:
+
+    .. code-block:: php
+
+        use App\Factory\PostFactory;
+
+        /**
+         * @dataProvider postDataProvider
+         */
+        public function test_post_via_data_provider(PostFactory $factory, string $expectedTitle): void
+        {
+            $factory->create();
+
+            $factory::assert()->exists(['title' => $expectedTitle]);
+        }
+
+        public static function postDataProvider(): iterable
+        {
+            yield from PostFactory::new()->sequence(
+                [
+                    ['title' => 'foo'],
+                    ['title' => 'bar'],
+                ]
+            )->dataProvider(
+                [
+                    ['foo'],
+                    ['bar'],
+                ]
```

Comment:
not sure about this, since it can create data providers difficult to reason about...

Reply:
Yeah, you're right. Let's drop the `dataProvider()` for now and in the docs, show how to manually create a provider:

```php
        public static function postDataProvider(): iterable
        {
            $sequence = PostFactory::new()->sequence(
                [
                    ['title' => 'foo'],
                    ['title' => 'bar'],
                ]
            );

            foreach ($sequence as $factory) {
                yield [$factory];
            }
        }
```